src/marker.rs

```diff
@@ -826,6 +826,13 @@ impl<'py> Python<'py> {
     }
 
     /// Registers the object in the release pool, and tries to downcast to specific type.
+    #[cfg_attr(
+        not(feature = "gil-refs"),
+        deprecated(
+            since = "0.21.0",
+            note = "`Python::checked_cast_as` will be replaced by `PyObject::downcast_bound` in a future PyO3 version"
```

Comment:
Given the semantics of the "replacement" is slightly different, maybe something like the following wording is more helpful to users:

```suggestion
            note = "part of the deprecated GIL Ref API; use `obj.downcast_bound::<T>(py)` instead of `py.cast_as::<T>(obj)` to migrate"
```

What do you think of this (and similar below)?

Reply:
Sounds good